### PR TITLE
Do not expose data urls in errors

### PR DESCRIPTION
**Pull request by mitsuhiko**
@getsentry/infrastructure

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/3752)

### CODE REVIEW

File: src/sentry/lang/javascript/processor.py [hunk 1]
```diff
@@ -663,7 +673,7 @@ def expand_frames(self, frames, release):
                     else:
                         all_errors.append({
                             'type': EventError.JS_MISSING_SOURCE,
-                            'url': force_bytes(abs_path, errors='replace'),
+                            'url': expose_url(abs_path),
```
Comment (benvinegar) [inline src/sentry/lang/javascript/processor.py]: Seems only a matter of time that someone will create an error without using `expose_url` ... what if we funneled through a helper function?

Comment (benvinegar) [inline src/sentry/lang/javascript/processor.py]: i.e. can this be done in `EventError`?

**Comment by benvinegar** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3752)*

File: src/sentry/lang/javascript/processor.py [hunk 2]
```diff
@@ -66,6 +66,16 @@ class ZeroReturnError(Exception):
 logger = logging.getLogger(__name__)
 
 
+def expose_url(url):
+    if url is None:
+        return u'<unknown>'
+    if url.startswith('data:'):
+        return u'<data url>'
```
Comment (dcramer) [inline src/sentry/lang/javascript/processor.py]: maybe "data:<redacted>" would be better?

minor, and i dont care that much (esp since its merged already)

**Pull request merged**